tests/integration/extenders/ViewTest.php

```diff
@@ -36,4 +36,30 @@ public function custom_view_namespace_can_be_added_by_extender()
 
         $this->assertEquals('<html><body>Hello World!</body></html>', trim($this->app()->getContainer()->make(Factory::class)->make('integration.test::test')->render()));
     }
+
+    /**
+     * @test
+     */
+    public function can_replace_view_namespace_by_extender()
+    {
+        $this->extend(
+            (new Extend\View)
+                ->replace('flarum', dirname(__FILE__, 3).'/fixtures/views')
+        );
+
+        $this->assertEquals('<html><body>Hello World!</body></html>', trim($this->app()->getContainer()->make(Factory::class)->make('flarum::test')->render()));
+    }
+
+    /**
+     * @test
+     */
+    public function can_prepend_view_namespace_by_extender()
+    {
+        $this->extend(
+            (new Extend\View)
+                ->prepend('flarum', dirname(__FILE__, 3).'/fixtures/views')
+        );
+
+        $this->assertEquals('<html><body>Hello World!</body></html>', trim($this->app()->getContainer()->make(Factory::class)->make('flarum::test')->render()));
+    }
 }
```

Comment:
What's the difference between these two? Can we add tests showing that `prepend` keeps the originals for other views, whereas `replace` replaces them?

Reply:
They both keep them, at least when trying replace on the Flarum namespace, which is why I'm very confused.

Reply:
Here's the problem:

https://github.com/flarum/core/blob/a57ef5a3d500f15be2253843209b5726cb36b22e/vendor/illuminate/support/ServiceProvider.php#L156-L177

The way we attach the Flarum paths is we [register the namespace on boot](https://github.com/flarum/core/blob/a57ef5a3d500f15be2253843209b5726cb36b22e/src/Frontend/FrontendServiceProvider.php#L147-L147), by calling the above function. But that uses `callAfterResolving`, which IIRC means that the namespace is added after the replace/prepend calls.

Reply:
Is there any way to easily fix that, or do you think it might be better to keep it like this, and mention so in the docblock.

IMO, this way results in someone being less likely to mess up Flarum, and someone can still remove all the views through the `extend()` method, no?

Reply:
Is there any way to easily fix that, or do you think it might be better to keep it like this, and mention so in the docblock.

IMO, this way results in someone being less likely to mess up Flarum, and someone can still remove all the views through the `extend()` method, no?

Reply:
isn't `prepend` enough for the use case we're dealing with though, I think we should just use introduce that and if there's ever a use case for a full replacement, we can worry about it then.

Reply:
Fair enough, that sounds good to me.